### ISSUE TITLE
1.0.1RC1 - Unusable RAR File, extracts fine in 0.7.20

### ISSUE THREAD

**Issue by albino1** [sabnzbd/sabnzbd#560]
I'm running a fresh 1.0.1RC1 install on 64bit Debian Wheezy and I'm getting an `Unusable RAR File` on a very large NZB.  I know that it works, since it would have been reported from the source otherwise, so I ran it on my 0.7.20 Windows 7 install and it extracted just fine.  So I ran it a second time in 1.0.1RC1 only to end up with the same error.

Googling the error, it seems that it used to pop up quite frequently with RAR5 files before SABnzbd supported them, but RAR5 has been supported for ages now, so I don't think that's the case.

Interestingly, even though it reports a failure in the GUI, there's no actual error message in the log that I can find:

```
2016-04-30 05:51:30,831::INFO::[postproc:281] Starting PostProcessing on Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group => Repair:True, Unpack:True, Delete:True, Script:None, Cat:*
2016-04-30 05:51:30,831::INFO::[postproc:579] Par2 check starting on Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group
2016-04-30 05:51:31,599::INFO::[__init__:898] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/__verified__ missing
2016-04-30 05:51:31,600::INFO::[postproc:600] Running repair on set 85c8e39f584d16923d7bc2c43c31QR
2016-04-30 05:51:31,631::INFO::[newsunpack:1031] Quick-check for 85c8e39f584d16923d7bc2c43c31QR is OK, skipping repair
2016-04-30 05:51:31,632::INFO::[newsunpack:1104] Deleting /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/85c8e39f584d16923d7bc2c43c31QR.par2
2016-04-30 05:51:31,633::INFO::[postproc:627] Par2 check finished on Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group
2016-04-30 05:51:31,635::INFO::[misc:808] Creating directories: /home/username/Downloads/complete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group
2016-04-30 05:51:31,638::INFO::[postproc:353] Running unpack_magic on Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group
2016-04-30 05:51:31,651::INFO::[newsunpack:229] Unrar starting on /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group
2016-04-30 05:51:31,659::INFO::[newsunpack:450] Extracting rarfile /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/85c8e39f584d16923d7bc2c43c31QR.part001.rar (belonging to 85c8e39f584d16923d7bc2c43c31QR) to /home/username/Downloads/complete/_UNPACK_Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group
2016-04-30 05:51:31,660::INFO::[newsunpack:505] Got a password set by user: 123456789
2016-04-30 05:51:31,660::INFO::[newsunpack:563] Archive /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/85c8e39f584d16923d7bc2c43c31QR.part001.rar probably has full encryption
2016-04-30 05:51:31,773::INFO::[newsunpack:233] Unrar finished on /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group
2016-04-30 05:51:31,774::INFO::[postproc:358] unpack_magic finished on Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group
2016-04-30 05:51:31,782::INFO::[postproc:551] Cleaning up Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group (keep_basic=True)
2016-04-30 05:51:31,782::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_SpRVzi removed
2016-04-30 05:51:31,782::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_0__Z0z removed
2016-04-30 05:51:31,782::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_1Rp5TH removed
2016-04-30 05:51:31,782::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_eZOUPV removed
2016-04-30 05:51:31,782::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_ioP3PA removed
2016-04-30 05:51:31,782::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_p2eLWj removed
2016-04-30 05:51:31,783::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_HfHJjD removed
2016-04-30 05:51:31,783::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_tZox2_ removed
2016-04-30 05:51:31,783::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_29Y3EK removed
2016-04-30 05:51:31,783::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_1kX3Hu removed
2016-04-30 05:51:31,783::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_Wkc9nw removed
2016-04-30 05:51:31,783::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_0gO719 removed
2016-04-30 05:51:31,783::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_HwhDHx removed
2016-04-30 05:51:31,783::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_FVzrRw removed
2016-04-30 05:51:31,783::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_QKEUxg removed
2016-04-30 05:51:31,783::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_3aUv8i removed
2016-04-30 05:51:31,784::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_03Ej2A removed
2016-04-30 05:51:31,784::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_KKaKII removed
2016-04-30 05:51:31,784::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_CrS1zO removed
2016-04-30 05:51:31,784::INFO::[__init__:932] /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__/SABnzbd_nzf_1y4h1g removed
2016-04-30 05:51:31,789::INFO::[misc:1285] Cannot remove folder /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__
2016-04-30 05:51:31,789::INFO::[misc:1285] Cannot remove folder /home/username/Downloads/incomplete/Title.YEAR.COMPLETE.BLURAY.iNTERNAL-Group/__ADMIN__
2016-04-30 05:51:31,789::INFO::[postproc:83] Saving postproc queue
2016-04-30 05:51:31,790::INFO::[__init__:941] Saving data for postproc2.sab in /home/username/.sabnzbd/admin/postproc2.sab
2016-04-30 05:51:31,792::INFO::[downloader:277] Post-processing finished, resuming download
2016-04-30 05:51:32,792::INFO::[nzbqueue:250] Saving queue
2016-04-30 05:51:32,793::INFO::[__init__:941] Saving data for queue10.sab in /home/username/.sabnzbd/admin/queue10.sab
2016-04-30 05:51:32,793::INFO::[__init__:941] Saving data for totals10.sab in /home/username/.sabnzbd/admin/totals10.sab
2016-04-30 05:51:32,793::INFO::[__init__:941] Saving data for rss_data.sab in /home/username/.sabnzbd/admin/rss_data.sab
2016-04-30 05:51:32,794::INFO::[__init__:941] Saving data for watched_data2.sab in /home/username/.sabnzbd/admin/watched_data2.sab
2016-04-30 05:51:32,794::INFO::[postproc:83] Saving postproc queue
2016-04-30 05:51:32,794::INFO::[__init__:941] Saving data for postproc2.sab in /home/username/.sabnzbd/admin/postproc2.sab
2016-04-30 05:51:32,794::INFO::[postproc:680] Queue has finished, launching: None (None)
```

Additional information: it is a password protected RAR and the password contains both `%` and `$` if that matters.

I've sent the NZB to bugs@sabnzbd.org so you can look at it.  Thanks.

**Comment by shypike** [comment]
Did you try to run the Debian  unrar tool on the RAR file?
Which version is it?
The problem is, we rely on whatever unrar tool is installed on your system.

**Comment by albino1** [comment]
Yep, that's the problem.  For some reason I thought SABnzbd came already packaged with a recent unrar.  Apologies, I'll close the issue.  Thanks.

**Comment by albino1** [comment]
I can't seem to figure out how to upgrade the unrar in Debian Wheezy, if anybody has any suggestions let me know.

Current version `UNRAR 4.10 freeware`

Line ran: `apt-get install unrar`

Output:

```
Reading package lists... Done
Building dependency tree       
Reading state information... Done
unrar is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

After some heavy Googling I added the non-free updates, but it gives the same results.  FWIW, I ran `apt-get update` and `apt-get upgrade` before `apt-get install unrar`.
